### PR TITLE
refactor: rework stories from MDX to MDX + CSF

### PR DESCRIPTION
**Pull request by uladzimirdev**
### Description

new versions of storybook do not support MDX1 format of stories that we use, so I reworked them to MDX2-3 + CSF
https://storybook.js.org/docs/api/csf

in this PR we 
- applied codemods from https://storybook.js.org/docs/migration-guide#storiesmdx-to-mdxcsf 
- manually deleted dead code
- cleaned up render functions in stories to avoid naming collisions
- converted stories to tsx

### How to verify

- run storybook locally - it should work as before
- CI is green

### CODE REVIEW

File: frontend/src/metabase/ui/components/inputs/MultiSelect/MultiSelect.stories.js [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+  args,
+  argTypes,
+};
+
+export const Default_ = {
+  render: Default,
+  name: "Default",
+};
+
+export const EmptyMd_ = {
```
Comment (ranquild) [inline frontend/src/metabase/ui/components/inputs/MultiSelect/MultiSelect.stories.js:152]: Probably this indirection is no longer needed, and you could have `EmptyMd` without `_` and set `VariantTemplate` in `render`

**Review by ranquild — APPROVED** [review]
Fine if this is done by a codemod. It would be great to remove some odd code and convert to TS.

**Review by romeovs — APPROVED** [review]
Played around with some of the changed stories and all seems well. Amazing work!

File: frontend/src/metabase/ui/components/inputs/MultiSelect/MultiSelect.stories.js (hunk 1 above)
Comment (uladzimirdev) [reply frontend/src/metabase/ui/components/inputs/MultiSelect/MultiSelect.stories.js]: should be fixed in f039d27be2c98c2d4b55cc3dd98e65679bd8c9c3

**Comment by uladzimirdev** [comment]
@ranquild stories were cleaned up (`StoryName_` -> `StoryNam`e, removed `.bind({})` etc) and converted to tsx

**Pull request merged**